game/theater/theatergroundobject.py

```diff
@@ -481,18 +502,77 @@ def __init__(
             dcs_identifier="AA",
             sea_object=False,
         )
-        # Set by the SAM unit generator if the generated group is compatible
-        # with Skynet.
-        self.skynet_capable = False
+        self.aa_ranges: List[AirDefenseRange] = [aa_range]
+
+    # Returns the name of one single group
+    # It also changes the group name to a skynet readable
+    # If the skynet conditions are true (skynet_capability_for_range)
+    def single_group_name(self, group_index: int) -> str:
+
+        if self.is_skynet_capable(group_index):
+            # Return regular name if not skynet_capable
+            # Depends on the skynet_capability_for_range and
+            # if the search radar of the unit is alive
+            return self.skynet_group_name(group_index)
```

Comment:
This seems like something that should be done at campaign generation time rather than here, which is effectively how the current implementation operates. What am I missing?